src/kvstore.c

```diff
@@ -391,6 +392,8 @@ unsigned long long kvstoreScan(kvstore *kvs, unsigned long long cursor,
     int skip = !d || (skip_cb && skip_cb(d));
     if (!skip) {
         _cursor = dictScan(d, cursor, scan_cb, privdata);
+        /* In dictScan, scan_cb may delete entries (e.g., in active expire case). */
+        freeDictIfNeeded(kvs, didx);
```

Comment:
I think this can also cause an issue in kvstoreIteratorNext.

Reply:
right... It seems a little more troublesome now, although we can check the dictionary and release it in kvstoreIteratorRelease

Reply:
> kvstoreIteratorRelease

Sounds like the best tradeoff to me. The other option would be a refcount, so that we could free it when all the references were deleted. That seems like a lot of more memory and work though.